components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/View.kt

```diff
@@ -83,6 +85,69 @@ fun View.hideKeyboard() {
     imm.hideSoftInputFromWindow(windowToken, 0)
 }
 
+/**
+ * Attempts to call immersive mode using the View to hide the status bar and navigation buttons.
+ */
+fun View.enterToImmersiveMode() {
+    val context = this.context
+    if (context is Activity) {
+        val window = context.window
+        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+        window.decorView.systemUiVisibility = (View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+            or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+            or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+            or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+            or View.SYSTEM_UI_FLAG_FULLSCREEN
+            or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
+    }
+}
+
+/**
+ * Attempts to come out from immersive mode using the View.
+ */
+fun View.exitImmersiveModeIfNeeded() {
+    val context = this.context
+    if (context is Activity) {
+        if (WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON and context.window.attributes.flags == 0) {
+            // We left immersive mode already.
+            return
+        }
+
+        val window = context.window
+        window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+        window.decorView.systemUiVisibility =
+            View.SYSTEM_UI_FLAG_LAYOUT_STABLE or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+    }
+}
+
+/**
+ * Helper extensions that make enabling visible/gone of UI components easier if they are depended
+ * on boolean logic.
+ *
+ * Examples:
+ * ```
+ * toolbar.visibleOrGone(foo != bar)
+ * imageView.visibleOrGone(foo.isEnabled)
+ * ```
+ */
+fun View.visibleOrGone(enabled: Boolean) {
+    this.visibility = if (enabled) View.VISIBLE else View.GONE
+}
```

Comment:
It was mostly for product teams that needed to change multiple View's visibility dependent on some single logic.
Example:
```kotlin
if (enabled) {
  view1.visibility = View.VISIBLE
  view2.visibility = View.VISIBLE
  view3.visibility = View.VISIBLE
} else {
  view1.visibility = View.GONE
  view2.visibility = View.GONE
  view3.visibility = View.GONE
}
```
Instead we could do just this:
```kotlin
view1.visibleOrGone(enabled)
view2.visibleOrGone(enabled)
view3.visibleOrGone(enabled)
```
It's fair though to remove this as well because I can't see many use cases (for example) in focus where we follow that pattern.